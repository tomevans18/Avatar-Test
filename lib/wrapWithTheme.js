import { ThemeConsumer } from 'styled-components';
import theme from '../styles/theme';

export default (method, children, options, themeObject = theme) => {
  // eslint-disable-next-line no-underscore-dangle
  ThemeConsumer._currentValue = themeObject;
  return method(children, options);
};
