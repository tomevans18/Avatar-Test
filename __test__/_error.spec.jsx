import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import wrapWithTheme from '../lib/wrapWithTheme';

import ErrorPage from '../pages/_error';

let props;

describe('ErrorPage', () => {
  it('should render 404 as expected', async () => {
    props = await ErrorPage.getInitialProps({
      res: {
        statusCode: 404,
      },
    });
    const wrapper = wrapWithTheme(mount, <ErrorPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render other error as expected', async () => {
    props = await ErrorPage.getInitialProps({
      err: {
        statusCode: 500,
      },
    });
    const wrapper = wrapWithTheme(mount, <ErrorPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
