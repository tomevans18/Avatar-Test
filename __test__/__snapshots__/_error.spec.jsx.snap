// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render 404 as expected 1`] = `
.c0 {
  position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + 1.25em);
}

@supports (padding:max(0px)) {
  .c0 {
    padding: 0 max(1.25em,env(safe-area-inset-left));
  }
}

<Error
  statusCode={404}
>
  <ContentWrapper>
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  ",
                [Function],
                " position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + ",
                [Function],
                ");

  @supports (padding: max(0px)) {
    padding: 0 max(",
                [Function],
                ", env(safe-area-inset-left));
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <div>
            <h2>
              PAGE NOT FOUND
            </h2>
            <p>
              The page you are looking for might have been removed, or had its name changed. If you typed in the page address, please check that it's spelt correctly.
            </p>
            <p>
              Alternatively, go to the
               
              <Component
                href="/"
              >
                <a>
                  homepage
                </a>
              </Component>
               
              and follow the links to the information you want.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styled.div>
  </ContentWrapper>
</Error>
`;

exports[`ErrorPage should render other error as expected 1`] = `
.c0 {
  position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + 1.25em);
}

@supports (padding:max(0px)) {
  .c0 {
    padding: 0 max(1.25em,env(safe-area-inset-left));
  }
}

<Error
  statusCode={500}
>
  <ContentWrapper>
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  ",
                [Function],
                " position: relative;
  width: 100%;
  max-width: 67rem;
  margin: 0 auto;
  box-sizing: border-box;
  padding: 0 calc(env(safe-area-inset-right) + ",
                [Function],
                ");

  @supports (padding: max(0px)) {
    padding: 0 max(",
                [Function],
                ", env(safe-area-inset-left));
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <div>
            <h2>
              OOPS...
            </h2>
            <p>
              It appears that something went horribly wrong on our end. Please try again later.
            </p>
            <p>
              Alternatively, go to the
               
              <Component
                href="/"
              >
                <a>
                  homepage
                </a>
              </Component>
               
              and follow the links to the information you want.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styled.div>
  </ContentWrapper>
</Error>
`;
